### ISSUE TITLE
Add graphics API for drawing lines, the same Flash API.

### ISSUE THREAD

**Issue by alevys** [Gamua/Starling-Framework#1010]
Really need the ability to draw lines as in classic flash. Need native capability of the library. External libraries like starling-extension-graphics is already abandoned and will not be updated to match the current development of the Starling.

**Comment by PrimaryFeather** [comment]
It's definitely not a straightforward feature to add, but I know it would be super useful, and it's been on the back of my mind for a long time. Thanks for adding the issue!

I changed the title a little since I'd be adding lines as part of a more general drawing API, and I wouldn't force myself to stick 100% with the Flash API.

**Comment by alevys** [comment]
Do not have to repeat the flash API, the main thing that it were possible to draw. Very, very'll wait for this feature. :)

**Comment by 22Round** [comment]
+1

**Comment by hardcoremore** [comment]
+1

**Comment by matrix3d** [comment]
http://blog.bwhiting.co.uk/?p=423

**Comment by soimy** [comment]
@alevys Actually, there is a usable [starling-extension-graphics](https://github.com/StarlingGraphics/Starling-Extension-Graphics/tree/master/extension2.0/src/starling/display) for starling v2 which will be working after some minor tweaks. ( And here is my [fork](https://github.com/soimy/Starling-Extension-Graphics-2))

I believe there are more effective ways to do vector triangulation, and no Ex-version ( editable vector ). But at least it's working.

**Comment by Dima107** [comment]
> @alevys Actually, there is a usable [starling-extension-graphics](https://github.com/StarlingGraphics/Starling-Extension-Graphics/tree/master/extension2.0/src/starling/display) for starling v2 which will be working after some minor tweaks. ( And here is my [fork](https://github.com/soimy/Starling-Extension-Graphics-2))
> 
> I believe there are more effective ways to do vector triangulation, and no Ex-version ( editable vector ). But at least it's working.
> 
> Below is what I've done with the help of this extension:
> [![Gear&Chain Simulator](https://camo.githubusercontent.com/aeaf830ffa5acac59ffc32ff6dc189de619d4520/687474703a2f2f696d672e796f75747562652e636f6d2f76692f734a3652426d5f724d4f342f302e6a7067)](http://www.youtube.com/watch?v=sJ6RBm_rMO4)

I need starling grpahics extention too. But It looks like "beginTextureFill" and "lineTexture" doesn't works in oringal and your fork of graphic.
Can you help me with it?

**Comment by hardcoremore** [comment]
Yes texture fill does not work as well. I would like to have that but don't know how to implement it. I also have my own port here https://bitbucket.org/hardcoremore/starlinggraphicsextension/src/master/

**Comment by Fancy2209** [comment]
> @alevys Actually, there is a usable [starling-extension-graphics](https://github.com/StarlingGraphics/Starling-Extension-Graphics/tree/master/extension2.0/src/starling/display) for starling v2 which will be working after some minor tweaks. ( And here is my [fork](https://github.com/soimy/Starling-Extension-Graphics-2))
> 
> I believe there are more effective ways to do vector triangulation, and no Ex-version ( editable vector ). But at least it's working.
> 
> Below is what I've done with the help of this extension:
> [![Gear&Chain Simulator](http://img.youtube.com/vi/sJ6RBm_rMO4/0.jpg)](http://www.youtube.com/watch?v=sJ6RBm_rMO4 "Gear&Chain Simulator")

Is the code you used to triangulate the vectors available anywhere? I've been looking into trying to allow HW Vectors in flash and I'd love to see how other people have done it for ideas

**Comment by hardcoremore** [comment]
Just want to add in my fork: https://bitbucket.org/hardcoremore/starlinggraphicsextension/src/master/

Texture fill is working. Also you can draw lines with texture. I have also implemented miter, bevel and round joint types. But still there are some things to be fixed.

**Comment by PrimaryFeather** [comment]
Great to see an update on this, @hardcoremore! 😄 Thanks for sharing!

**Comment by Fancy2209** [comment]
> It's definitely not a straightforward feature to add, but I know it would be super useful, and it's been on the back of my mind for a long time. Thanks for adding the issue!
> 
> I changed the title a little since I'd be adding lines as part of a more general drawing API, and I wouldn't force myself to stick 100% with the Flash API.

What would you do different from the Flash API?
I'm looking into adding Texture Fills and Strokes to Canvas, so would love to know what you'd like in Starling vs what is in Flash
